src/components/Motorcycle/Details/index.js

```diff
@@ -0,0 +1,213 @@
+import {
+  Link, useNavigate, useParams,
+} from 'react-router-dom';
+import ReactTimeAgo from 'react-time-ago';
+import { useToasts } from 'react-toast-notifications';
+import { deleteMotorcycle, fetchMotorCycleAsync } from '../../../Redux/actions/motorcycle';
+import { cancelReservation, fetchReservationsAsync } from '../../../Redux/actions/reservation';
+import responsive from '../../../utils/responsive';
+import AddReservation from '../../Reservation/AddReservation';
+import AllContainer from '../../Shared/AllContainer';
+import UserTemplate from '../../Shared/UserTemplate';
+import '../styles/index.css';
+
+function MotorCycleDetails() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const {
+    data, loading, error, createdBy, reservations,
+  } = useSelector((state) => state.motorcycles?.motorcycle);
+  const { currentUser, loading: userLoading, token } = useSelector((state) => state.account);
+  const [reserved, setReserved] = useState(false);
+  const [isCreator, setCreator] = useState(false);
+  const {
+    data: Reservations,
+    loading: reservationLoading,
+  } = useSelector((state) => state.reservations);
+  const dispatch = useDispatch();
+  const { addToast } = useToasts();
+  const [show, showReservation] = useState(false);
+  useEffect(() => {
+    if (data?.id !== id) {
+      dispatch(fetchMotorCycleAsync(id));
+    }
+  }, [id]);
+  useEffect(() => {
+    if (!Reservations.length) { dispatch(fetchReservationsAsync()); }
+  }, []);
+  const [image, setImage] = useState(data?.images[0]);
+  useEffect(() => setImage(data?.images[0]), [data?.images]);
+  useEffect(() => {
+    const reserved = Reservations.find((r) => (r?.user_id == currentUser?.id.toString())
+    && (r.motorcycle_id == id));
+    setReserved(reserved);
+  }, [Reservations, data]);
+  useEffect(() => {
+    if (data?.user_id === currentUser?.id) { setCreator(true); }
+  }, [data, currentUser]);
+  const DeleteMotorcycle = () => {
+    if (confirm('Are you sure you want to delete this motorcycle?. All reservations are also canceled')) {
+      dispatch(deleteMotorcycle(id, token, addToast, navigate));
+    }
+  };
+  return (
+    <AllContainer navigation={false} loading={loading && reservationLoading} error={error} data={data ? ['1'] : []}>
+      <div className="container">
+        <div className="row">
+          <div className="col-lg-9">
+            <Link to="/">
+              <h1 className="font-italic logo display-4 text-dark">Bcycom</h1>
+            </Link>
+            <div className="mt-4">
+              <img src={image} alt="" className="w-100" />
+            </div>
+
+          </div>
+          <div className="col-lg-3" style={{ height: '100vh' }}>
+            <div className="flex-column justify-content-around d-flex align-items-center w-100" style={{ height: '100%' }}>
+              <div className="d flex flex-column justify-content-start" style={{ fontFamily: "'Rubik', sans-serif" }}>
```

Comment:
See the previous component and apply as see necessary anywhere else who similar issue